### PR TITLE
Allow client only channels

### PR DESCRIPTION
**Pull request by mmihic**
Also makes channel api symmetric with net/http

R: @abhinav / @weikai77

### CODE REVIEW

File: golang/channel.go [hunk 1]
```diff
@@ -21,12 +21,23 @@ package tchannel
...
+
+	"golang.org/x/net/context"
+)
+
+var (
+	errAlreadyListening = errors.New("channel already listening")
+)
+
+const (
+	clientOnlyHostPort = "0.0.0.0:0"
```
Comment (abhinav) [inline golang/channel.go]: IMO, ephemeral might be a better term for this than client-only.

File: golang/channel.go [hunk 2]
```diff
@@ -21,12 +21,23 @@ package tchannel
...
 	"os"
 	"path/filepath"
+	"sync"
 	"time"
+
+	"golang.org/x/net/context"
+)
+
+var (
+	errAlreadyListening = errors.New("channel already listening")
```
Comment (abhinav) [inline golang/channel.go]: Don't you want this to be a publicly accessible value so that users can match on it?

**Comment by abhinav** [comment]
Looks good besides minor issues mentioned above.

File: golang/channel.go (hunk 2 above)
Comment (mmihic) [inline golang/channel.go]: This is a programming error, so not something you'd branch on.  Usually logic errors are kept unexported.

**Pull request merged**